### PR TITLE
Trim EOL Ruby and Rails versions out of the matrix

### PR DESCRIPTION
**Pull request by petergoldstein**

### CODE REVIEW

**Review by adamstegman — COMMENTED** [review]
Thanks! Build here: https://github.com/apokalipto/devise_saml_authenticatable/actions/runs/1668993003

Can you also remove the old Gemfiles from `spec/support`?

**Comment by adamstegman** [comment]
Oh, there's a couple version checks in `idp_template.rb` and `sp_template.rb` that can be removed now too!

**Comment by petergoldstein** [comment]
@adamstegman Ok, this is ready for review - all of the obsolete version checks are removed and it's green.

File: spec/support/bin/bundle [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
```
Comment (adamstegman) [inline spec/support/bin/bundle:1]: Do we need this file?

Comment (petergoldstein) [reply spec/support/bin/bundle]: No.  It's a side effect of running tests manually locally.  I've deleted and added to .gitignore.

**Comment by petergoldstein** [comment]
@adamstegman I expect everything to come out green.  So feel free to review again (I've removed that file)

**Review by adamstegman — APPROVED** [review]

**Comment by rmoehn** [comment]
Just because it's end of life doesn't mean that nobody is using it anymore… The version check for `skip_before_filter :verify_authenticity_token` is still relevant to us. Of course, I can monkey-patch the skip_before_filter call in. But it cost me some troubleshooting time to figure out what is going on. A note in the README or the wiki would have been great.

**Comment by adamstegman** [comment]
@rmoehn thanks for sharing your experience. I'm sorry we put you in a tricky spot to debug. I'll share the same response I did [over here](https://github.com/apokalipto/devise_saml_authenticatable/pull/222#issuecomment-1113420217), but regarding Rails instead of Ruby. 🙂 

> Using Ruby versions older than 2.6 carries a security risk, just like using an old version of ruby-saml that we don't support. For me, it also means fewer possibilities for flaky test as we carry a huge matrix of test versions.

I know upgrading Rails is much more difficult than upgrading Ruby version, unfortunately. Are there features you depend on in 1.8.0 or 1.9.0 of this gem that are forcing you to upgrade and patch?

**Comment by rmoehn** [comment]
I totally agree with the flaky tests point. They can be a big drain. And no, other than the filter thing, the gem works great with Rails 4.

By the way, I'm impressed with the way you take care of this repo. Always friendly, always replying with something useful, even to grumpy comments like mine. Thank you!

**Comment by rmoehn** [comment]
Would it be helpful if I draft a note for the README or the wiki about the issue with Rails 4?

**Comment by adamstegman** [comment]
Thanks! A suggestion or PR would definitely help me—wherever it would have helped you to see it will probably help someone else.

**Comment by rmoehn** [comment]
I'll come up with something by Monday.

**Comment by rmoehn** [comment]
I've added two pages to the wiki:
- https://github.com/apokalipto/devise_saml_authenticatable/wiki/Can%27t-verify-CSRF-token-authenticity-%28Rails-4-and-below%29
- https://github.com/apokalipto/devise_saml_authenticatable/wiki/Redirect-loop

**Comment by adamstegman** [comment]
Thank you! Those look excellent.

**Pull request merged**